lib/hook/producers/github.ex

```diff
@@ -0,0 +1,29 @@
+defmodule Accent.Hook.Producers.GitHub do
+  use GenStage
+
+  def start_link do
+    GenStage.start_link(__MODULE__, :ok, name: __MODULE__)
+  end
+
+  def init(:ok) do
+    {:producer, {:queue.new(), 0}, dispatcher: GenStage.BroadcastDispatcher}
+  end
+
+  def handle_call({:external_document_update, event}, from, {queue, demand}) do
+    dispatch_events(:queue.in({from, event}, queue), demand, [])
+  end
+
+  def handle_demand(incoming_demand, {queue, demand}) do
+    dispatch_events(queue, incoming_demand + demand, [])
+  end
+
+  defp dispatch_events(queue, demand, events) do
+    with d when d > 0 <- demand,
```

Comment:
What does `d` represent in this case?

Reply:
`d when d > 0 <- demand`

It simply represents the `demand` binding. If it were named `demand` instead of the single letter, it would rebind `demand`, which is unnecessary.

This code was taken from `GenStage` source as the `QueueBroadcaster` example. It has since been updated to use the default `handle_demand/3` which handles the case of `demand < 0` 😄 I’ve updated the module with the `QueueBroadcaster` example on `gen_stage` `master` branch.